v2/features/farms/flows/zap/ZapFlow.tsx

```diff
@@ -0,0 +1,260 @@
+import { useMachine } from "@xstate/react";
+import { Chains, PickleModelJson } from "picklefinance-core";
+import {
+  ChainNativetoken,
+  UserBrineryData,
+  UserTokenData,
+} from "picklefinance-core/lib/client/UserModel";
+import { LightningBoltIcon } from "@heroicons/react/solid";
+
+import { AppDispatch } from "v2/store";
+import Button from "v2/components/Button";
+import Modal from "v2/components/Modal";
+import { AssetWithData, BrineryWithData, CoreSelectors, JarWithData } from "v2/store/core";
+import { stateMachine, Actions, States } from "../stateMachineUserInput";
+import Form from "./Form";
+import { jarDecimals } from "v2/utils/user";
+import AwaitingConfirmation from "../deposit/AwaitingConfirmation";
+import AwaitingReceipt from "../AwaitingReceipt";
+import Success from "../Success";
+import Failure from "../Failure";
+import { useJarContract, useTransaction } from "../hooks";
+import { TransferEvent } from "v1/containers/Contracts/Jar";
+import { UserActions } from "v2/store/user";
+import { formatDollars, truncateToMaxDecimals } from "v2/utils";
+import { eventsByName, getNativeName } from "../utils";
+import { isAcceptingDeposits } from "v2/store/core.helpers";
+import { useSelector } from "react-redux";
+import { TokenSelect } from "../deposit/FormUniV3";
+import { solidityKeccak256 } from "ethers/lib/utils";
+
+type DepositType = "jar" | "brinery";
+
+interface Props {
+  asset: AssetWithData | BrineryWithData;
+  balances: UserTokenData | UserBrineryData | undefined;
+  nativeBalances: ChainNativetoken | undefined;
+  type: DepositType;
+}
+
+const getZapTokens = (
+  userNativeData: ChainNativetoken | undefined,
+  userTokenData: UserTokenData | undefined,
+  asset: AssetWithData,
+  core: PickleModelJson.PickleModelJson | undefined,
+): any => {
+  const componentTokens = userTokenData?.componentTokenBalances || undefined;
+
+  const jarChain = core?.chains.find((chain) => chain.network === asset.chain);
+
+  const nativeName = jarChain ? getNativeName(jarChain.gasTokenSymbol) : undefined;
+  const wrappedName = nativeName && "W" + nativeName;
+
+  const nativeDecimals =
+    core?.tokens.find((x) => x.chain === asset.chain && x.id === nativeName)?.decimals || 18;
+
+  return {
+    ...(!componentTokens
+      ? {}
+      : Object.keys(componentTokens).reduce((acc, curr) => {
+          const currDecimals =
+            core?.tokens.find((x) => x.chain === asset.chain && x.id === curr.toLowerCase())
+              ?.decimals || 18;
+
+          return {
+            ...acc,
+            [curr.toUpperCase()]: {
+              balance: componentTokens[curr]?.balance,
+              allowance: componentTokens[curr]?.allowance, // TODO: this is only the jar allowance (used in univ3 zaps), need to update for specific zap contract/protocol in pf-core. Can potentially use the same property to express allowance for respective protocol's zap contract, since there's no overlap between univ3 jars and zaps currently
```

Comment:
@bcstryker as described, a component token's allowance for a protocol's zap contract should be configured here in pf-core